onnxruntime/python/onnxruntime_pybind_sparse_tensor.cc

```diff
@@ -0,0 +1,427 @@
+        // as base object for numpy array to make sure rep python object does not go away
+        // while numpy array view is in use.
+        // See https://github.com/pybind/pybind11/issues/2271
+        py::array result(indices.Shape().GetDims(), indices.Data<int64_t>(), py::cast(*view));
+        assert(!result.owndata());
+        // Set a read-only flag
+        PyArray_CLEARFLAGS(reinterpret_cast<PyArrayObject*>(result.ptr()), NPY_ARRAY_WRITEABLE);
+        return result;
+      });
+
+  py::class_<PySparseCsrView>(m, "SparseCsrView")
+      .def("inner", [](const PySparseCsrView* view) -> py::array {
+        const auto& indices = view->Inner();
+        // Since rep is a registered pybind object, it will create an extra reference
+        // as base object for numpy array to make sure rep python object does not go away
+        // while numpy array view is in use.
+        // See https://github.com/pybind/pybind11/issues/2271
+        py::array result(indices.Shape().GetDims(), indices.Data<int64_t>(), py::cast(*view));
+        assert(!result.owndata());
+        // Set a read-only flag
+        PyArray_CLEARFLAGS(reinterpret_cast<PyArrayObject*>(result.ptr()), NPY_ARRAY_WRITEABLE);
+        return result;
+      })
+      .def("outer", [](const PySparseCsrView* view) -> py::array {
+        const auto& indices = view->Outer();
+        // Since rep is a registered pybind object, it will create an extra reference
+        // as base object for numpy array to make sure rep python object does not go away
+        // while numpy array view is in use.
+        // See https://github.com/pybind/pybind11/issues/2271
+        py::array result(indices.Shape().GetDims(), indices.Data<int64_t>(), py::cast(*view));
+        assert(!result.owndata());
+        // Set a read-only flag
+        PyArray_CLEARFLAGS(reinterpret_cast<PyArrayObject*>(result.ptr()), NPY_ARRAY_WRITEABLE);
+        return result;
+      });
+
+  py::class_<PySparseBlockSparseView>(m, "SparseBlockSparseView")
+      .def("index", [](const PySparseBlockSparseView* view) -> py::array {
+        const auto& indices = view->Index();
+        // Since rep is a registered pybind object, it will create an extra reference
+        // as base object for numpy array to make sure rep python object does not go away
+        // while numpy array view is in use.
+        // See https://github.com/pybind/pybind11/issues/2271
+        py::array result(indices.Shape().GetDims(), indices.Data<int64_t>(), py::cast(*view));
+        assert(!result.owndata());
+        // Set a read-only flag
+        PyArray_CLEARFLAGS(reinterpret_cast<PyArrayObject*>(result.ptr()), NPY_ARRAY_WRITEABLE);
+        return result;
+      });
+
+  py::class_<PySparseTensor>(m, "SparseTensor")
+      // Factor method to create a COO Sparse Tensor from numpy arrays acting as backing storage.
+      // Numeric arrays memory is used as is with reference count increment. All other supported
```

Comment:
nit: 'Factory' method. 

There are a few places where this change should be applied.